cli/server/server.go

```diff
@@ -336,8 +340,20 @@ Main:
 		case err := <-errChan:
 			shutdownErr = fmt.Errorf("server error: %w", err)
 			cancel()
-
+		case sig := <-sighupCh:
+			switch sig {
+			case syscall.SIGHUP:
+				log.Info("SIGHUP received, restarting rpc-server")
+				serverErr := rpcServer.Shutdown()
+				if serverErr != nil {
+					errChan <- fmt.Errorf("error while restarting rpc-server: %w", serverErr)
+					break
+				}
+				rpcServer = server.New(chain, cfg.ApplicationConfiguration.RPC, serv, serv.GetOracle(), log)
```

Comment:
This way config won't be reread.

Reply:
Yes, that's exactly the way we'd like it to work. IIUC, we don't want to reload the config, we want to re-read the cert files with old names (the files may be updated by this moment).